lib/galaxy/webapps/galaxy/controllers/dataset.py

```diff
@@ -361,8 +361,8 @@ def set_edit(self, trans, payload=None, **kwd):
                     annotation = sanitize_html(payload.get("annotation"))
                     self.add_item_annotation(trans.sa_session, trans.get_user(), data, annotation)
                 # if setting metadata previously failed and all required elements have now been set, clear the failed state.
-                if data._state == trans.model.Dataset.states.FAILED_METADATA and not data.missing_meta():
-                    data._state = None
+                if data.state == trans.model.Dataset.states.FAILED_METADATA and not data.missing_meta():
```

Comment:
Given the PR title just making sure you want this here.

Reply:
Look at that, I guess search and replace led me to save a pending change ... I didn't mean to, but this is the right thing to do.